### PR TITLE
tests: fix unstable test_node_compact_after_delete

### PR DESCRIPTION
**Pull request by Little-Wallace**
Signed-off-by: Little-Wallace <bupt2013211450@gmail.com>

### What problem does this PR solve?

Issue Number: close https://github.com/tikv/tikv/issues/3217 

Problem Summary:

### What is changed and how it works?

What's Changed:
I fix an unstable test `test_node_compact_after_delete`, see more details in https://github.com/tikv/tikv/issues/3217.

And I port a new feature to `fail_point`, which make us can set callback at `fail_point` in tests. I use this feature solve the problem of unstable test, because we often  can not ensure how long shall we wait in a test. With callback, we can make `fail_point` notify test-thread.

### Related changes

- PR to update `tikv/fail-rs`: https://github.com/tikv/fail-rs/pull/50

### Check List 

- Unit test

### CODE REVIEW

File: components/raftstore/src/store/worker/compact.rs [hunk 1]
```diff
@@ -169,6 +169,7 @@ impl Runnable<Task> for Runner {
                                 );
                             }
                         }
+                        fail_point!("raftstore::compact:CheckAndCompact");
```
Comment (zhangjinpeng1987) [inline components/raftstore/src/store/worker/compact.rs]: Maybe `raftstore::compact::CheckAndCompact::AfterCompact` is better?

Comment (Little-Wallace) [reply components/raftstore/src/store/worker/compact.rs]: Ok. I will fix it. And I think it is better that other fail-point also be named as this. `{{module}}::{{file}}::{{method}}::{{point}}`.

**Comment by Little-Wallace** [comment]
@zhangjinpeng1987 PTAL

**Pull request merged**